src/containers/MyNdla/Learningpath/EditLearningpathStepsPage.tsx

```diff
@@ -67,16 +68,53 @@ export const EditLearningpathStepsPage = () => {
   const onSaveStep = async (values: FormValues) => {
     if (data?.myNdlaLearningpath?.id) {
       const transformedData = formValuesToGQLInput(values);
-      await createStep({
+      const res = await createStep({
         variables: {
           learningpathId: data.myNdlaLearningpath.id,
           params: { ...transformedData, language: i18n.language, showTitle: false },
         },
       });
-      setIsCreating(false);
+      if (!res.errors?.length) {
+        setIsCreating(false);
+        setTimeout(() => document.getElementById(`edit-step-${res.data?.newLearningpathStep?.id}`)?.focus(), 0);
+      }
     }
   };
 
+  const onDeleteStep = async (stepId: number, close: VoidFunction) => {
+    if (data?.myNdlaLearningpath?.id) {
+      const currentStepIndex = data.myNdlaLearningpath.learningsteps.findIndex(({ id }) => stepId === id);
+      const res = await deleteStep({
+        variables: {
+          learningstepId: stepId,
+          learningpathId: data.myNdlaLearningpath.id,
+        },
+      });
+
+      if (!res.errors?.length) {
+        close();
+        const nextStep =
+          data.myNdlaLearningpath.learningsteps?.[currentStepIndex + 1] ??
+          data.myNdlaLearningpath.learningsteps?.[currentStepIndex - 1];
+
+        if (nextStep) {
+          setTimeout(() => document.getElementById(`edit-step-${nextStep?.id}`)?.focus(), 0);
+        } else {
+          document.getElementById(SKIP_TO_CONTENT_ID)?.focus();
+        }
+      }
+    }
+  };
+
+  const onOpenCreate = async () => {
+    setIsCreating(true);
+    setTimeout(() => document.getElementById("create-step-form")?.getElementsByTagName("input")?.[0]?.focus(), 500);
```

Comment:
Dette var eneste løsningen jeg kom på for å håndtere lazy loadingen av komponenten? Er det kanskje mulig å lazyloade kun richTextEditor? Eller var det dumt å gjøre?

Reply:
Nei, det holder dessverre ikke å kun lazyloade RichTextEditor. Serialisering og deserialisering skjer utenfor RichTextEditor, og drar inn alt av slate-pakker.